Fix sporadic test failure in t/io/eintr.t (+ cleanup)

bram-perl opened the pull request:
The test would sporadically fail; troubleshooting this revealed that
the size of the pipe buffer could vary between pipes which this code
didn't take into account.

What the code did:

1. Create two pipes ($in, $out)
2. Get buffer size of '$out' pipe
3. Do something that should block (using the buffer size)
4. Create two new pipes ($in, $out)
5. Do something that should block (using the buffer size)

The problem: the pipe created in 4. does not always have the same buffer size as the pipe created in 1.

Adding some debug info in the test:

	# pipe buf size: 8193
	ok 4 - read/die: read status
	ok 5 - read/die: close status
	# pipe buf size: 65537
	not ok 6 - print/close: sig handler close status

During test 6 it reused the smaller buffer size and wrote data
to the new pipe. It expected this to block but the buffer of the pipe
used for test 6 is 65536 bytes. End result: the pipe buffer is not
full (only 8194 bytes were written) so the test does not block which
causes the alarm not to trigger which causes the test failure.

Fix all this by reinitialising `$pipe_buf_size` when creating new pipes.

(Observed by @khwilliamson , diagnosed with help of @wolfsage )

+

Apply some small cleanup to t/io/eintr.t

Commit:
t/io/eintr.t: Rename variable to `$pipe_buf_size`

Rename the `$surely_this_arbitrary_number_is_fine` to `$pipe_buf_size`
since it's no longer an arbitrary number.

Commit:
t/io/eintr.t: Whitespace fixes

Most of the code was intended with 4 spaces but some of it was
intended with tab. Fix the inconsistenty and replace tabs with spaces.

Commit:
t/io/eintr.t: Fix sporadic test failure

The test would sporadically fail; troubleshooting this revealed that
the size of the pipe buffer could vary between pipes which this code
didn't take into account.

What the code did:
	1. Create two pipes ($in, $out)
	2. Get buffer size of '$out' pipe
	3. Do something that should block (using the buffer size)
	4. Create two new pipes ($in, $out)
	5. Do something that should block (using the buffer size)

The problem: the pipe created in 4. does not always have the same
             buffer size as the pipe created in 1.

Adding some debug info in the test:
	# pipe buf size: 8193
	ok 4 - read/die: read status
	ok 5 - read/die: close status
	# pipe size: 65537
	not ok 6 - print/close: sig handler close status

During test 6 it reused the smaller buffer size and wrote data
to the new pipe. It expected this to block but the buffer of the pipe
used for test 6 is 65536 bytes. End result: the pipe buffer is not
full (only 8194 bytes were written) so the test does not block which
causes the alarm not to trigger which causes the test failure.

Fix all this by reinitialising `$pipe_buf_size` when creating new pipes.

(Observed by @khwilliamson, diagnosed with help of @wolfsage)

Commit:
t/io/eintr.t: Add `diag` when F_GETPIPE_SZ failed

When the fcntl call to get the size of the pipe buffer failed then
it would silently fallback to 0xfffff.

-> Make this fallback no longer silent and add a diagnostic message
   when it happens.

Leont commented on t/io/eintr.t:
```diff
@@ -55,24 +55,39 @@ if ($^O eq 'VMS' || $^O eq 'MSWin32' || $^O eq 'cygwin' || $^O =~ /freebsd/ || $
+    };
+    if ($@ or not $pipe_buf_size) {
+        $pipe_buf_size = 0xfffff;
+        diag("fcntl F_GETPIPE_SZ failed, falling back to $pipe_buf_size");
```
I don't think this should be a diag, as it's not an issue that warrents user intervention. It should probably be a note instead.

bram-perl replied:
If you insist on `note()` then I will change it but I'm a bit in doubt;

With `note()` the message is only shown when it's being run in verbose mode..
But if the code gets the pipe buf size wrong then it could lead to test failures and with `note()` the output is then **not** include the report making it harder to troubleshoot/debug (since it may not always happen...)

With `diag()`: if the o/s doesn't support F_GETPIPE_SZ (are there any?) then it will always show the message leading to cluttered output..


(I suppose ideally the diag message would be shown when there was a test failure but that would be slightly trickier..)

Leont replied:
> I suppose ideally the diag message would be shown when there was a test failure

Yeah, I think that's the better way to look at it. One shouldn't diag unless things are failing.

Commit:
t/io/eintr.t: only show diag message on failure

Also improve the error message by capturing the exception/error
returned by the `fnctl` call.

khwilliamson commented on t/io/eintr.t:
```diff
@@ -73,18 +73,27 @@ sub fresh_io {
         $pipe_buf_size = 0xfffff;
-        diag("fcntl F_GETPIPE_SZ failed, failing back to $pipe_buf_size");
+        $pipe_buf_err = "fcntl F_GETPIPE_SZ failed" . ($err ? " ($err)" : "") .
+                        ", failing back to $pipe_buf_size";
```
I don't understand "failing back"

demerphq replied:
its a fancy way of saying "defaulting to $pipe_buf_size". :-)
